packages/core/test/unit/route/openapi/createSchemaDefinition.test.ts

```diff
@@ -390,4 +390,49 @@ describe('createSchemaDefinition', () => {
 			}
 		});
 	});
+
+	it('should support dependencies', () => {
+		@openapi.definition({
+			title: 'Customer',
+			oneOf: [
+				{ required: ['a'] },
+				{ required: ['b'] }
+			]
+		})
+		class Customer {
+			@openapi.prop()
+			a?: string;
+
+			@openapi.prop()
+			b?: string;
+
+			@openapi.prop()
+			@openapi.dependsOn(['a'])
```

Comment:
my only worry here is that, if we want to support all the json schema options, we may need to add a lot of decorators.
The other option would be for the `@definition` decorator to accept all the json schema options (as you did before)